### PR TITLE
[SPARK-15456][PYSPARK] Fixed PySpark shell context initialization when HiveConf not present

### PR DESCRIPTION
**Pull request by BryanCutler**
## What changes were proposed in this pull request?

When PySpark shell cannot find HiveConf, it will fallback to create a SparkSession from a SparkContext.  This fixes a bug caused by using a variable to SparkContext before it was initialized.
## How was this patch tested?

Manually starting PySpark shell and using the SparkContext

### CODE REVIEW

**Comment by BryanCutler** [comment]
cc @andrewor14

**Comment by andrewor14** [comment]
oops, LGTM

File: python/pyspark/shell.py [hunk 1]
```diff
@@ -44,9 +44,9 @@
         .enableHiveSupport()\
         .getOrCreate()
 except py4j.protocol.Py4JError:
-    spark = SparkSession(sc)
+    spark = SparkSession(SparkContext.getOrCreate())
```
Comment (andrewor14) [inline python/pyspark/shell.py]: actually, we should use the builder pattern here instead: `SparkSession.builder.getOrCreate()`

Comment (BryanCutler) [inline python/pyspark/shell.py]: sure, that looks better

**Comment by andrewor14** [comment]
Thanks merging into master 2.0

**Pull request closed without merging**